internal/installercache/installercache.go

```diff
@@ -177,41 +268,49 @@ func (i *Installers) evict() {
 			return nil
 		}
 
-		//save the other files based on their mod time
-		files.Add(&fileInfo{path, info})
-		totalSize += info.Size()
+		includeFile := false
+		stat, ok := info.Sys().(*syscall.Stat_t)
+		if !ok {
+			i.log.Warnf("could not determine hardlink status for %s - item will not be filtered", path)
+			includeFile = true
+		}
+		if stat.Nlink == 1 { // This means we have no hardlinks to the file.
+			includeFile = true
+		}
+
+		// if the file has no hardlinks then add it to the queue
+		if includeFile {
+			files.Add(&fileInfo{path, info})
+			totalSize += info.Size()
+		}
+
 		return nil
 	}
 
-	err := filepath.Walk(i.cacheDir, visit)
+	err := filepath.Walk(i.config.CacheDir, visit)
 	if err != nil {
 		if !os.IsNotExist(err) { //ignore first invocation where the cacheDir does not exist
-			i.log.WithError(err).Errorf("release binary eviction failed to inspect directory %s", i.cacheDir)
+			i.log.WithError(err).Errorf("release binary eviction failed to inspect directory %s", i.config.CacheDir)
 		}
-		return
+		return false
 	}
 
-	//prune the hard links just in case the deletion of resources
-	//in ignition.go did not succeeded as expected
-	for idx := 0; idx < len(links); idx++ {
-		finfo := links[idx]
-		//Allow a grace period of 5 minutes from the link creation time
-		//to ensure the link is not being used.
-		grace := time.Now().Add(DeleteGracePeriod).Unix()
-		if finfo.info.ModTime().Unix() < grace {
-			os.Remove(finfo.path)
-		}
-	}
+	// TODO: We might want to consider if we need to do this longer term, in theory every hardlink should be automatically freed.
+	i.pruneExpiredHardLinks(links, linkPruningGracePeriod)
 
-	//delete the oldest file if necessary
-	for totalSize >= int64(float64(i.storageCapacity)*CacheLimitThreshold) {
+	// delete the oldest file if necessary
+	evicted := false
+	for i.shouldEvict(totalSize); i.shouldEvict(totalSize) && files.Len() > 0; {
```

Comment:
Pretty sure you don't need the three clause for loop here any more since you're not initializing a variable or incrementing anything.

```suggestion
	for i.shouldEvict(totalSize) && files.Len() > 0 {
```